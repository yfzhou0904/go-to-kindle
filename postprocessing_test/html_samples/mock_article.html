<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Mock Article for Testing</title>
</head>
<body>
    <article>
        <header>
            <h1>Mock Article for Testing</h1>
            <p class="meta">Published on January 1, 2025</p>
        </header>

        <main>
            <p>This is a mock article designed to test various image processing scenarios in our post-processing pipeline.</p>

            <h2>Image Test Cases</h2>

            <!-- Case 1: Base64 embedded image (should work in all scenarios) -->
            <p>Base64 embedded image:</p>
            <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg==" alt="Base64 test image" width="100" height="100" />

            <!-- Case 2: Relative URL image (should be downloaded when baseURL is provided) -->
            <p>Relative URL image:</p>
            <img src="/test_images/sample.png" alt="Sample PNG" width="400" height="300" loading="lazy" srcset="/test_images/sample.png 1x, /test_images/sample@2x.png 2x" />

            <!-- Case 3: Absolute URL image (should be downloaded) -->
            <p>Absolute URL image:</p>
            <img src="https://example.com/test_images/sample.jpg" alt="Sample JPEG" width="500" height="400" sizes="(max-width: 768px) 100vw, 50vw" />

            <!-- Case 4: Image with query parameters -->
            <p>Image with query parameters:</p>
            <img src="/test_images/sample.gif?v=123&size=medium" alt="Sample GIF with params" width="200" height="200" />

            <!-- Case 5: Picture element with source tags -->
            <picture>
                <source srcset="/test_images/sample.webp" type="image/webp" />
                <source srcset="/test_images/sample.png" type="image/png" />
                <img src="/test_images/sample.png" alt="Picture element test" />
            </picture>

            <!-- Case 6: Image without src (should be removed) -->
            <img alt="No src image" width="100" height="100" />

            <!-- Case 7: Image with empty src (should be removed) -->
            <img src="" alt="Empty src image" />

            <!-- Case 8: Image with WordPress-specific attributes (attrs should be cleaned) -->
            <img src="/test_images/sample.png"
                 alt="WordPress style image"
                 data-orig-size="800,600"
                 data-medium-file="/test_images/sample-medium.png"
                 data-large-file="/test_images/sample-large.png"
                 width="400"
                 height="300" />
        </main>

        <footer>
            <p>This mock article tests the image processing pipeline comprehensively.</p>
        </footer>
    </article>
</body>
</html>
