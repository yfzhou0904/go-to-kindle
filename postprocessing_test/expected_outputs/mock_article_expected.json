{
  "testName": "mock_article",
  "inputFile": "html_samples/mock_article.html",
  "baseURL": "https://example.com/article",
  "excludeImages": false,
  "expectedImageCount": 8,
  "mustNotContain": [
    "srcset",
    "loading",
    "sizes",
    "data-orig-size",
    "data-medium-file",
    "data-large-file",
    "width=\"",
    "height=\""
  ],
  "mustContain": [
    "This is a mock article designed to test",
    "data:image/png;base64,",
    "Image Test Cases",
    "Base64 embedded image:",
    "WordPress style image"
  ]
}
